File: typescript/codify/templates/abx-python/src/lib/resource.py
import json


class Resource:

    def __init__(self, context, input):
        self.context = context
        self.input = input

    def create(self):
        ''' Create a new resource '''

        return {
            'status': f"Resource '{self.input.get('name')}' created"
        }

    def remove(self):
        ''' Remove existing resource '''

        return {
            'status': f"Resource '{self.input.get('name')}' removed"
        }

    def make_internal_request(self):
        ''' Create an internal GET reqest and return the response. '''

        data = self.context.request('/deployment/api/deployments', 'GET', None)
        return json.loads(data['content'])

    def make_external_request(self):
        ''' Create an external POST reqest and return the response. '''

        payload = {
            'userId': 1,
            'title': 'My TODO item',
            'completed': False
        }
        data = self.context.request(
            'https://jsonplaceholder.typicode.com/todos', 'POST', payload)
        return json.loads(data['content'])

File: typescript/codify/templates/abx-javascript/src/lib/resource.js


class Resource {
  constructor(context, input) {
    this.context = context;
    this.input = input;
  }

  /**
   * Create a new resource.
   * @returns operation status
   */
  async create() {
    return {
      status: `Resource ${this.input.name} created`,
    };
  }

  /**
   * Remove existing resource.
   * @returns operation status
   */
  async remove() {
    return {
      status: `Resource ${this.input.name} removed`,
    };
  }

  /**
   * Create an internal GET reqest and return the response.
   * @returns the response
   */
  async makeInternalRequest() {
    const data = await this.context.request('/deployment/api/deployments', 'GET');
    return data.content;
  }

  /**
   * Create an external POST request and return the response.
   * @returns the response
   */
  async makeExternalRequest() {
    const payload = {
      userId: 1,
      title: 'My TODO item',
      completed: false,
    };
    const data = await this.context.request('https://jsonplaceholder.typicode.com/todos', 'POST', payload);
    return data.content;
  }
}

exports.Resource = Resource;
